README.md

```diff
@@ -92,51 +93,101 @@ app.get('/', async function (req, reply) {
+Persisted Queries solve this problem by having the client send a generated ID, instead of the full query string, resulting in a smaller request. The server can use an internal lookup to turn this back into a full query and return the result.
 
+The `persistedQuerySettings` option lets you configure this for Fastify GQL. There are a few default options available, included in `GQL.PersistedQueryDefaults`.
+
+#### Prepared
+
+Prepared queries give the best performance in all use cases, at the expense of tooling complexity. Queries must be hashed ahead of time, and a matching set of hashes must be available for both the client and the server. Additionally, version control of query hashes must be considered, e.g. queries used by old clients may need to be kept such that hashes can be calculated at build time. This can be very useful for non-public APIs, but is impractical for public APIs.
+
+Clients can provide a full query string, or set the `persisted` flag to true and provide a hash instead of the query in the request:
 ```js
-'use strict'
+{
+  query: '<hash>',
+  persisted: true
+}
+```
 
-const Fastify = require('fastify')
+A map of hashes to queries must be provided to the server at startup:
+```js
 const GQL = require('fastify-gql')
 
-const app = Fastify()
-
-const schema = `
-  type Query {
-    add(x: Int, y: Int): Int
-  }
-`
+app.register(GQL, {
+  ...
+  persistedQuerySettings: GQL.PersistedQueryDefaults.Prepared({
+    '<hash>':  '{ add(x: 1, y: 1) }'
+  })
+})
+```
 
-const resolvers = {
-  Query: {
-    add: async (_, { x, y }) => x + y
-  }
-}
+Alternatively the `peristedQueries` option may be used directly, which will be internally mapped to the `Prepared` default:
+```js
+const GQL = require('fastify-gql')
 
 app.register(GQL, {
-  schema,
-  resolvers,
+  ...
   persistedQueries: {
-    '<unique-hash>': '{ add(x: 1, y: 1) }'
+    '<hash>':  '{ add(x: 1, y: 1) }'
   }
 })
+```
 
-app.listen(3000)
+#### PreparedOnly
+
+This offers similar performance and considerations to the `Prepared` queries, but only allows persisted queries. This provides additional secuirity benefits, but means that the server **must** know all queries ahead of time or will reject the request.
+
+The API is the same as the `Prepared` default.
+
+Alternatively the `peristedQueries` and `onlyPersisted` options may be used directly, which will be internally mapped to the `PreparedOnly` default:
+```js
+const GQL = require('fastify-gql')
+
+app.register(GQL, {
+  ...
+  persistedQueries: {
+    '<hash>': '{ add(x: 1, y: 1) }'
+  },
+  onlyPersisted: true
+})
 ```
 
-### Disallowing unknown queries
+#### Automatic
+
+This default is compatible with `apollo-client`, and requires no additional tooling to set up at the cost of some performance. In order for this mode to be effective, you must have long lived server instances (i.e *not* cloud functions). This mode is also appropriate for public APIs where queries are not known ahead of time.
```

Comment:
Adding some links to the relevant portion of apollo-client docs would be really handy, thanks.

Reply:
Sure